beacon-chain/db/kv/blocks.go

```diff
@@ -153,12 +153,9 @@ func (k *Store) BlockRoots(ctx context.Context, f *filters.QueryFilter) ([][]byt
 func (k *Store) HasBlock(ctx context.Context, blockRoot [32]byte) bool {
 	ctx, span := trace.StartSpan(ctx, "BeaconDB.HasBlock")
 	defer span.End()
-	k.blocksLock.RLock()
-	if blk, exists := k.blocks[blockRoot]; exists && blk != nil {
-		k.blocksLock.RUnlock()
+	if v := blockCache.Get(string(blockRoot[:])); v != nil {
```

Comment:
why does the cache have keys as strings ? Anything wrong with bytes

Reply:
The cache API requires string keys unfortunately

Reply:
maybe we can look at it in another PR